bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/events/ThingEventFactory.java

```diff
@@ -328,7 +331,8 @@ public static ChannelTriggeredEvent createTriggerEvent(String event, ChannelUID
         checkNotNull(event, "event");
 
         String topic = buildTopic(CHANNEL_TRIGGERED_EVENT_TOPIC, channelUID);
-        TriggerEventPayloadBean bean = new TriggerEventPayloadBean(event, channelUID.getAsString());
+        TriggerEventPayloadBean bean = new TriggerEventPayloadBean(event,
+                AbstractEvent.buildSource(THING_SOURCE, channelUID.getAsString()));
```

Comment:
are you sure about this one?

Reply:
you're right, I'm setting the channel UID here, _not_ the source. will fix